python/hail/context.py

```diff
@@ -255,39 +254,197 @@ def import_gen(self, path, sample_file=None, tolerance=0.2, npartitions=None, ch
+        
+        .. note::
+        
+            Use ``delimiter='\\s+'`` to specify whitespace delimited files.
+            
+        The ``comment`` is an optional parameter which causes Hail to skip any line that starts in the
+        given pattern. Passing ``comment='#'`` will skip any line beginning in a pound sign, for example.
+        
+        The ``missing`` parameter defines the representation of missing data in the table. 
+        .. note::
+        
+            The ``comment`` and ``missing`` parameters are **NOT** regexes.
+
+        The ``no_header`` option indicates that the file has no header line. If this option is passed, 
+        then the column names will be ``f0``, ``f1``, ... ``fN`` (0-indexed). 
+        
+        The ``types`` option allows the user to pass the types of columns in the table. This is a 
+        dict keyed by ``str``, with :py:class:`~hail.expr.Type` values. See the examples above for
+        a standard usage. Additionally, this option can be used to override type imputation. For example,
+        if a column in a file refers to chromosome and does not contain any sex chromosomes, it will be
+        imputed as an integer, while most Hail methods expect chromosome to be passed as a string. Using
+        the ``impute=True`` mode and passing ``types={'Chromosome': TString()}`` will solve this problem.
+        
+        The ``min_partitions`` option can be used to increase the number of partitions (level of sharding)
+        of an imported table. The default partition size depends on file system and a number of other 
+        factors (including the ``min_block_size`` of the hail context), but usually is between 32M and 128M.
+        
+        :param paths: files to import.
+        :type paths: str or list of str
 
         :param key: List of key columns.
         :type key: str or list of str
 
-        :param npartitions: Number of partitions.
-        :type npartitions: int or None
-
-        :param config: Configuration options for importing text files
-        :type config: :class:`.TextTableConfig`
-
+        :param min_partitions: Minimum number of partitions.
+        :type min_partitions: int or None
+
+        :param bool no_header: File has no header and the N columns are named ``_0``, ``_1``, ... ``_N`` (0-indexed)
+        
+        :param bool impute: Impute column types from the file
+        
+        :param comment: Skip lines beginning with the given pattern
+        :type comment: str or None
+        
+        :param str delimiter: Field delimiter regex
+        
+        :param str missing: Specify identifier to be treated as missing
+        
+        :param types: Define types of fields in annotations files   
+        :type types: dict with str keys and :py:class:`.Type` values
+    
         :return: Key table constructed from text table.
         :rtype: :class:`.KeyTable`
         """
 
-        if not isinstance(key, list):
+        if isinstance(paths, list):
```

Comment:
God, so much code.  Can will kill this while we wait for python type checking?  It makes my eyes bleed.